doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,384 @@
+.. role:: html(raw)
+   :format: html
+
+Program-capture sharp bits and debugging tips
+=============================================
+
+Program-capture is PennyLane's new internal representation for hyrbid quantum-classical 
+programs that maintains the same familiar feeling frontend that you're used to, 
+while providing better performance, harmonious integration with just-in-time compilation 
+frameworks like `Catalyst <https://docs.pennylane.ai/projects/catalyst/en/stable/index.html>`__ 
+(:func:`~.pennylane.qjit`) and JAX-jit, and compact representations of programs that preserve 
+their structure.
+
+Program-capture in PennyLane is propped up by JAX's internal representation of programs 
+called `jaxpr <https://docs.jax.dev/en/latest/jaxpr.html>`__, which works by leveraging 
+the Python interpreter to extract the core elements of programs and capture them 
+into a light-weight language.
+
+In part because of PennyLane's NumPy-like syntax and functionality, quantum programs 
+written with PennyLane translate nicely into the language of jaxpr, letting your 
+quantum-classical programs burn the same fuel that powers JAX and just-in-time compilation.
+
+The quantum tape/script has been PennyLane's trusty program representation since 
+its inception, but our vision with program-capture is to supplant the quantum tape
+as the default program representation in PennyLane, and to support *more* than just 
+the core features of the PennyLane you have come to know and love. 
+
+There are some **quirks and restrictions to be aware of while we strive towards 
+that ideal**. Additionally, we've added backwards-compatibility features that make 
+the transition from tape-based code to program-capture be a smooth one. In this 
+document, we provide an overview of the constraints, "gotchas" to be aware of, and
+features that will help get your existing tape-based code working with program-capture.
+
+.. note::
+
+    #. From here onwards, :func:`~.pennylane.capture.enable` is assumed to be present 
+    within the scope of code examples, unless otherwise stated. This ensures that 
+    program-capture is enabled.
```

Comment:
For reference documentation, since it is more likely to be consulted than read end-to-end, I wouldn't actually mind repeating enable in the codeblocks!

It means we'll have examples that when copied and pasted into fresh notebooks will work as shown